### ISSUE TITLE
change of sign in surface loss

### ISSUE THREAD

**Issue by ghost** [LIVIAETS/boundary-loss#40]
i have use the surface loss that has been mentioned here " https://github.com/LIVIAETS/boundary-loss/issues/14#issuecomment-547007286"

so i am using a dataset where the images have two different labels, so i have trained two models for two different label for the first label the surface loss is positive in all the cases. whereas while using the second set of label,  the surface loss is negative. how can i solve this problem

**Comment by ghost** [comment]
thanks in advance

**Comment by HKervadec** [comment]
It is hard to help you, with so little information. 

I am not fully sure this is really a problemM with different labels, the optimal loss will have different values, and the problem will be easier (or harder) to optimize. 

So, if one set of labels ends up with a positive loss, it means that the network is having troubles to learn for that specific case. You could try to tune parameters of the optimizer (learning rate & co), the loss weight, and try another base loss (there are many alternatives to GDL).

Best,

Hoel
